parity/deprecated.rs

```diff
@@ -209,6 +209,22 @@ pub fn find_deprecated(args: &Args) -> Vec<Deprecated> {
 		result.push(Deprecated::Removed("--fast-and-loose"));
 	}
 
+	if args.cmd_dapp {
+		result.push(Deprecated::Removed("parity dapp"));
+	}
+
+	if args.arg_dapp_path.is_some() {
+		result.push(Deprecated::Removed("--dapp-path"));
+	}
+
+	if args.flag_no_dapps {
+		result.push(Deprecated::Removed("--no-dapps"));
+	}
+
+	if args.arg_dapps_path != "" {
```

Comment:
It looks like the default is `"$BASE/dapps"` so this deprecation might be always printing?

My suggestion would be to change `arg_dapps_path` in `parity/cli/mod.rs` to be `Option<String>` with default to `None`. In that case, we can detect whether there's any attempt to set that arg and reliably print out deprecation warnings.

Reply:
Ah yeah, true. I put it as an empty string because the `deprecated` test builds the config with `Default::default()` (which sets the value to empty).
Fixed with an `Option`.